server/gameconnection.py

```diff
@@ -294,14 +295,19 @@ async def handle_game_result(self, army, result):
         except (KeyError, ValueError):  # pragma: no cover
             self._logger.warning("Invalid result for %s reported: %s", army, result)
 
-    async def handle_operation_complete(self, army, secondary, delta):
-        if not int(army) == 1:
+    async def handle_operation_complete(
+        self,
+        primary_objectives_complete: typing.Union[bool, int],
+        secondary_objectives_complete: typing.Union[bool, int],
```

Comment:
So, the weird thing here is that the correct type is actually `str` for all parameters because that's how the message is serialized by the game afaik. That's also the reason why we have to call `int`, `bool`, etc on the values before we use them.

Reply:
Ah thanks, I wasn't sure how this was being serialized.